src/foundry/collections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,179 @@
+   */
+  static createCompendium(metadata: Compendium.Metadata, options?: Compendium.Settings): Promise<CompendiumCollection>;
+
+  constructor(metadata: Compendium.Metadata);
+
+  /**
+   * The compendium metadata which defines the compendium content and location
+   */
+  metadata: Compendium.Metadata;
+
+  /**
+   * A subsidiary collection which contains the more minimal index of the pack
+   */
+  index: foundry.utils.Collection<Compendium.IndexEntry>;
+
+  /**
+   * A debounced function which will clear the contents of the Compendium pack if it is not accessed frequently.
+   */
+  private _flush;
+
+  /**
+   * The canonical Compendium name - comprised of the originating package and the pack name
+   */
+  get collection(): string;
+
+  /**
+   * Access the compendium configuration data for this pack
+   */
+  get config(): Compendium.Settings;
+
+  /**
+   * Track whether the Compendium Collection is locked for editing
+   */
+  get locked(): boolean;
+
+  /**
+   * Track whether the Compendium Collection is private
+   */
+  get private(): boolean;
+
+  /**
+   * A convenience reference to the label which should be used as the title for the Compendium pack.
+   */
+  get title(): string;
+
+  /**
+   * Load the Compendium index and cache it as the keys and values of the Collection.
+   */
+  getIndex(): Promise<CompendiumCollection<T>['index']>;
+
+  /**
+   * Get a single Document from this Compendium by ID.
+   * The document may already be locally cached, otherwise it is retrieved from the server.
+   * @param id -              The requested Document id
+   * @returns     The retrieved Document instance
+   */
+  getDocument(id: string): Promise<T>;
+
+  /**
+   * Load multiple documents from the Compendium pack using a provided query object.
+   * @param query -           A database query used to retrieve documents from the underlying database
+   * @returns   The retrieved Document instances
+   */
+  getDocuments(query?: object): Promise<T[]>;
+
+  /**
+   * Import a Document into this Compendium Collection.
+   * @param document -    The existing Document you wish to import
+   * @returns   The imported Document instance
+   */
+  importDocument(document: T): Promise<T>;
+
+  /**
+   * Fully import the contents of a Compendium pack into a World folder.
+   * @returns    The imported Documents, now existing within the World
+   */
+  importAll({ folderId, folderName }?: { folderId?: string | null; folderName?: string }): Promise<T[]>;
+
+  /**
+   * Add a Document to the index, capturing it's relevant index attributes
+   * @param document -       The document to index
+   */
+  indexDocument(document: T): void;
+
+  /**
+   * Assign configuration metadata settings to the compendium pack
+   * @param settings -  The object of compendium settings to define
+   * @returns          A Promise which resolves once the setting is updated
+   */
+  configure(settings?: Compendium.Settings): Promise<any>;
```

Comment:
The return value can probably be typed more strictly. If anything, we could replace the `any` with an `unknown` for now and add a TODO.